src/v/kafka/server/replicated_partition.cc

```diff
@@ -302,6 +305,13 @@ raft::replicate_stages replicated_partition::replicate(
   model::record_batch_reader&& rdr,
   raft::replicate_options opts) {
     using ret_t = result<raft::replicate_result>;
+    if (_partition->is_read_replica_mode_enabled()) {
```

Comment:
makes sense to me. presumably we already protect against writing into read replicas somehow? cc @bharathv @Lazin

Reply:
FWIW we do this in the produce path directly: https://github.com/redpanda-data/redpanda/blob/bd9292c0825b2f7ab1551761d5b89ecb28d8667e/src/v/kafka/server/handlers/produce.cc#L415-L421

But this partition proxy that @mmaslankaprv introduced did a bunch of read replica checks at this level, so it seemed like the right place to do these checks, I guess doing this check here means the produce path one can be removed because it's redundant. Thoughts on where the right place is?

Related is the discussion towards the generic data path code.

Reply:
yeh moving it down into the partition layer makes sense.